### ISSUE TITLE
Questions about WAP

### ISSUE THREAD

**Issue by bilal2vec** [JianshuZhang/WAP#19]
Hi,

I was reimplementing WAP and had some questions.

1. Did you try using a resnet based encoder instead of using a vgg or densenet?
2. Did you calculate the BLEU score as an additional metric?
3. What are the train and val loss of the models after training?
4. Looking at your data iterator code, it looks like you don't resize images but group  together images with the same size into a batch. Is this correct?
5. In issue https://github.com/JianshuZhang/WAP/issues/8#issuecomment-392633715, you wrote that the 2013 ground truth latex labels aren't normalized. How much did the exprate decrease when training and validating on un-normalized data?
6. When training, do you use the ground truth label of the previous timestep as the input to the model at the current timestep (teacher forcing), or do you instead use the model's prediction at the previous timestep?

Thanks

**Comment by JianshuZhang** [comment]
sorry for the late.
1. densenet is best for CROHM dataset, if your training set is big enough, maybe resnet is better.
2. I didn't try BLEU.
3. You can see the training log.
4. Yes, I didn't resize images.
5. If your training latex and test latex are mis-match, the exprate will decrease a lot.
6. use the ground truth label of the previous timestep as the input

**Comment by bilal2vec** [comment]
Hi, thanks for the response

I tried training another WAP implementation (https://github.com/menglin0320/wap-on-tensor) on the new 2019 crohme dataset (https://www.cs.rit.edu/~crohme2019/task.html) but couldn't get the model to generalize to the new out of distribution validation set. Did you also experience problems with your model generalizing to the 2013 test set? If so, how were you able to solve that problem?

Thanks

**Comment by JianshuZhang** [comment]
Recently, we also take part in the CROHME2019 competition. Our model can be tested on CROHME 2013, 2014, 2016 dataset. So, I didn't meet your problems.

**Comment by Zhang-O** [comment]
@JianshuZhang 
1.densenet is best for CROHM dataset, if your training set is big enough, maybe resnet is better.
 how many pics will satisfy the ' big enough '? 20k will ok?

**Comment by JianshuZhang** [comment]
Yes, the encoder depends on your task, it would be easy to change the encoder part.
For densenet on CROHME dataset, less than 40k will be OK.

**Comment by Zhang-O** [comment]
thank you for your reply.  if I want to change the backbone to resnet, pics num should be more than 40k.Is my understanding right?
